Gbe 841: Show performer image as modal on /gbe

jonkiparsky opened the pull request.

gawaineo commented:
Tested. Ran pepdiff and navigated through web app to view the changes in on local server. Looks like it's ready to merge to me.

BenignCremator commented:
All tests pass, and code review looked great.  I like extracting reusable markup to form new templates.  Functional testing was....  kinda working.  The images are not sized to the popups, or the popups are not sized to the images.  Some of those images are HUGE, and run off the screen.  I am willing to approve once I talk to either @jon or @bethlakshmi about if anything and what should be done about that.

jonkiparsky commented:
Added max-size styling to image in modal.

jonkiparsky commented:
@BenignCremator @bethlakshmi @gawaineo This should be ready to go.

gawaineo commented:
@jonkiparsky when I try to change the picture to a bigger one with the dimensions: 1920 x 1080, I notice that it doesn't get uploaded `"POST /persona/edit/1345 HTTP/1.1" 302 0`. I get a 302 error and no message saying it wasn't uploaded.

Another interesting behavior that I noticed is when I uploaded another picture and it successfully saves, is that when I click the image to see the larger image in the modal, the profile image is alternating between the recently uploaded and the previously uploaded picture.

jonkiparsky commented:
This sounds like you're running into the caching issues. Do the problems persist if you stop your local server and restart?

bethlakshmi commented:
I'm pretty unhappy with the way the thumbnails display now.  I spent a fair amount of time in the profile picture thumbnail ticket trying to find an alignment that made different picture formats somewhat visible.  Is there a way you can pull all the white space around them when on the landing page (/gbe)?

Firefox looks way worse than Chrome - and I can't tell what the difference is - but even Chrome really down-scales the images past what the original intentions of the gbe_thumbnail were - I think because they are in a button?

Also, I'm getting a really messed up looking X on the modal box.  It's drifted down over the upper band at the top of the window.

Images from Chrome and Firefox on Mac attached.  I can't really try a decent version of Windows - I don't have one at present.  If you tell me what browser this looks OK in, I'll retest and/or argue further.  Since we don't have a baseline browser set for testing, I just tested with the two prevalent browsers that I had handy on my Mac.  

![firefox-modal-box](https://cloud.githubusercontent.com/assets/4542707/18497055/409a7aae-79f9-11e6-94e7-56426fc33f27.png)
![chrome-modal-box](https://cloud.githubusercontent.com/assets/4542707/18497054/4094fd54-79f9-11e6-9751-1e1e6ff66f5d.png)
![chrome - 9-13-2016](https://cloud.githubusercontent.com/assets/4542707/18497056/40a8c816-79f9-11e6-9c3e-b628e46ee331.png)
![firefox screen - 9-13-2016](https://cloud.githubusercontent.com/assets/4542707/18497057/40ad3284-79f9-11e6-90c0-d13be75d83cf.png)

gawaineo commented:
@jonkiparsky it still does it when I when I restarted the server, and when I first tested it was on a new machine. It's alternating between previous pic and newly selected picture.

bethlakshmi commented:
Hmm... I don't see the problems that @gawaineo is seeing.  I got this image:

http://themebin.files.wordpress.com/2010/03/monastery_1920x1080.jpg

Which I shamelessly stole from the Internet... 

... to work, and it rescales nicely to fit the modal.  I'm wondering if this is a performance thing, where the machine is timing out somehow and then hanging itself in cache issues?

bethlakshmi commented on expo/gbe/templates/gbe/profile/incl_performer.tmpl:
```diff
@@ -1,78 +1,26 @@
-	  <div class="gbe_no_pic">&nbsp;</div>
-	{% endif %}
-	  <div class="performer_text">
-          {% if performer.contact = profile %}
```
Where is the analog of this line in the refactored code?

BLOCKER:  With the change in the code, I can now edit troupes in which I am a *member* but not the owner.  In master, I can *view* troupes where I am a member.  This is a regression.

bethlakshmi commented:
Summary:
- unit tests work
- no need to pepdiff
- I could get over the bad visuals, if we push to test, get input from Scratch and fix via other tickets.
- I can't get over the regression I raised as a BLOCKER.

jonkiparsky commented:
@bethlakshmi fixed the regression

BenignCremator commented:
Potential BLOCKER:  now some of the names disappear in Ubuntu/Konqueror.  There were there last time I checked.  See attached screen grab, a closeup of the effected area.  Still willing to approve if the behavior is not seen elsewhere and we decided that Konqueror is not that important of a browser, or if we decide that this is best fixed in a separate ticket.
![gbe-missing-names](https://cloud.githubusercontent.com/assets/4823397/18563610/514a910a-7b56-11e6-8791-c3a61b86d004.jpg)

jonkiparsky commented:
Fixed Hunter's bug. In the process I realized that we don't actually have any working flows for combos, so I removed them from the page. There are no combos on prod as of this evening, so I don't think anyone will be very inconvenienced by this. 
For the time being, I think it best to just drop the "combo" idea until we can make some sense of it.

BenignCremator commented:
Code review looked good, functional tests look good, runcoverage looks good.  Approved, but not merging due to significant recent changes.  Want another reviewer.

gawaineo commented:
- Changes look good and the photos are consistent and centered.
- pep8 looks good as well
- test results from py.test are great
![samurai_shot](https://cloud.githubusercontent.com/assets/13838812/18893151/a506c34e-84da-11e6-85af-38bb556dd94c.png)
![lionel](https://cloud.githubusercontent.com/assets/13838812/18893183/c036b340-84da-11e6-9b4d-352054d43a55.png)

**Note:** In the screenshot below I notice only the top picture shows where you see Samurai and Samurai2. I believe @BenignCremator pointed this out last time (so maybe there is a separate ticket).


![2nd_photo_missing](https://cloud.githubusercontent.com/assets/13838812/18893132/8d6a910c-84da-11e6-90f0-f9f62a3a1fb6.png)

I approve as well, changes look great.

BenignCremator commented:
On 09/27/2016 06:05 PM, gawaineo wrote:

> *Note:* In the screenshot below I notice only the top picture shows
> where you see Samurai and Samurai2. I believe @BenignCremator
> <https://github.com/BenignCremator> pointed this out last time (so maybe
> there is a separate ticket).

Weird.  When I tested it locally, the missing photos had returned, and 
they were always in the same place before.  Wonder why this one is gone, 
different reason?  The previous ones were tied to troupes (evidenced by 
@jon fix in template), and this seems more like an individual.

Benign Cremator
